### PR TITLE
Feat/202/deploy local branch to aws

### LINKED ISSUE #202: Add option to deploy a local Git branch (of BigchainDB) to AWS
The AWS deployment scripts currently deploy BigchainDB using `pip install bigchaindb`, i.e. they use the latest version of the `bigchaindb` package in PyPI.

We need another option: the ability to deploy a local Git branch to AWS.

I am already started on the code for this.

### PR DESCRIPTION
**Pull request by ttmc**
The main thing this PR adds is the ability to deploy to AWS from a branch in your local Git repository (rather than just the latest `bigchaindb` package on PyPI). The associated documentation was also updated. From a user's point of view, the main change is a new third argument for `startup.sh`: that third argument can either be `pypi` or the name of the local Git branch.

Other changes:

When installing the "base software" (using `apt-get install`), install `python3-dev` and `python3-setuptools` rather than `python-dev` and `python3-pip`. This brings the installation more in line with the recommended installation procedure in the documentation. It also led to getting the latest `pip`, `wheel` and `setuptools` by adding the follwing lines to the fabfile:
```python
    sudo('easy_install3 pip')
    sudo('pip install -U pip wheel setuptools')
```

I also made some  minor formatting tweaks to `startup.sh` and `fabfile.py`.

This resolves #202

### CODE REVIEW

File: deploy-cluster-aws/startup.sh [hunk 1]
```diff
@@ -1,28 +1,33 @@
...
 fi
 
-if [ -z "$2" ]
-  then
+if [ -z "$2" ]; then
     printErr "<number_of_nodes_in_cluster>"
     exit 1
 fi
 
+if [ -z "$3" ]; then
```
Comment (vrde) [inline deploy-cluster-aws/startup.sh]: I would keep `$3` optional, with `pypi` as default value (this should be the most common case).

File: deploy-cluster-aws/fabfile.py [hunk 2]
```diff
@@ -111,11 +113,22 @@ def install_rethinkdb():
...
 
 
-# Install BigchainDB (from PyPI)
+# Install BigchainDB from PyPI
 @task
 @parallel
-def install_bigchaindb():
-    sudo('python3 -m pip install bigchaindb')
+def install_bigchaindb_from_pypi():
+    sudo('pip install bigchaindb')
```
Comment (r-marques) [inline deploy-cluster-aws/fabfile.py]: This should probably be `pip3 install bigchaindb`.
I think ubuntu already comes with pip for python2 installed by default so doing `pip install bigchaindb` will probably try to install bigchaindb with python2

File: deploy-cluster-aws/fabfile.py [hunk 3]
```diff
@@ -111,11 +113,22 @@ def install_rethinkdb():
...
+
+
+# Install BigchainDB from a Git archive file
+# named bigchaindb-archive.tar.gz
+@task
+@parallel
+def install_bigchaindb_from_git_archive():
+    put('bigchaindb-archive.tar.gz')
+    run('tar xvfz bigchaindb-archive.tar.gz')
+    sudo('pip install .')
```
Comment (r-marques) [inline deploy-cluster-aws/fabfile.py]: same as my previous comment

**Comment by diminator** [comment]
PR #212 (merged in current master) uses a new version of cryptoconditions (0.1.6) which doesnt depend on `libffi-dev` and `libssl-dev` anymore, since the cryptography package was removed from it's dependencies.

Can you update the fabfile dependencies to exclude these libraries?

File: deploy-cluster-aws/fabfile.py [hunk 4]
```diff
@@ -65,9 +65,11 @@ def install_base_software():
     sudo('apt-get -y -f install')
     sudo('apt-get -y install build-essential wget bzip2 ca-certificates \
                      libglib2.0-0 libxext6 libsm6 libxrender1 libssl-dev \
```
Comment (diminator) [inline deploy-cluster-aws/fabfile.py]: remove `libssl-dev` and `libffi-dev` here (and run a quick test)

**Comment by ttmc** [comment]
Okay I made all the suggested changes.

@diminator @vrde @r-marques --- please check that I did what you were thinking and then we should be good to merge this PR.

**Comment by r-marques** [comment]
:+1:

**Comment by diminator** [comment]
:+1:

**Comment by vrde** [comment]
:+1:

**Pull request merged**